activemodel/test/cases/error_test.rb

```diff
@@ -0,0 +1,196 @@
+# frozen_string_literal: true
+
+require "cases/helper"
+require "active_model/error"
+
+class ErrorTest < ActiveModel::TestCase
+  class Person
+    extend ActiveModel::Naming
+    def initialize
+      @errors = ActiveModel::Errors.new(self)
+    end
+
+    attr_accessor :name, :age
+    attr_reader   :errors
+
+    def read_attribute_for_validation(attr)
+      send(attr)
+    end
+
+    def self.human_attribute_name(attr, options = {})
+      attr
+    end
+
+    def self.lookup_ancestors
+      [self]
+    end
+  end
+
+  def test_initialize
+    base = Person.new
+    error = ActiveModel::Error.new(base, :name, :too_long, foo: :bar)
+    assert_equal base, error.base
+    assert_equal :name, error.attribute
+    assert_equal :too_long, error.type
+    assert_equal({ foo: :bar }, error.options)
+  end
+
+  test "initialize without type" do
+    error = ActiveModel::Error.new(Person.new, :name)
+    assert_equal :invalid, error.type
+    assert_equal({}, error.options)
+  end
+
+  test "initialize without type but with options" do
+    options = { message: "bar" }
+    error = ActiveModel::Error.new(Person.new, :name, options)
+    assert_equal :invalid, error.type
+    assert_equal(options, error.options)
+  end
+
+  # match?
+
+  test "match? handles mixed condition" do
+    subject = ActiveModel::Error.new(self, :mineral, :not_enough, count: 2)
+    assert_equal false, subject.match?(:mineral, :too_coarse)
+    assert_equal true, subject.match?(:mineral, :not_enough)
+    assert_equal true, subject.match?(:mineral, :not_enough, count: 2)
+    assert_equal false, subject.match?(:mineral, :not_enough, count: 1)
```

Comment:
Can we change these to `assert` and `assert_not`?  I don't think asserting a specific boolean return value is necessary (maybe we want to return `nil` or some truthy object later).